File: Manual/LightField/Add-in and Automation SDK/Samples/VisualBasic Add-Ins/Metadata Sample/MetaDataSample.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Markup
Imports System.AddIn
Imports System.IO
Imports System.AddIn.Pipeline
Imports PrincetonInstruments.LightField.AddIns

'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
'''''''''' Acquire multiple frames and show the meta data for each frame ''''''''''
'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
Public Class AddinMetaDataSample
    Inherits AddInBase

    Private control_ As MetadataSample

    Public ReadOnly Property UISupport As UISupport
        Get
            Return UISupport.ExperimentSetting
        End Get
    End Property

    Public Sub Activate(ByVal app As ILightFieldApplication)
        'Capture interface
        LightFieldApplication = app
        'Build your controls            
        control_ = New MetadataSample(LightFieldApplication)
        ExperimentSettingElement = control_
    End Sub

    Public Sub Deactivate()
    End Sub


    Public Overrides ReadOnly Property UIExperimentSettingTitle As String
        Get
            Return "Metadata Sample"
        End Get
    End Property

End Class

File: Manual/LightField/Add-in and Automation SDK/Samples/CSharp Add-Ins/Metadata Sample/MetadataSample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.AddIn;
using System.AddIn.Pipeline;

using PrincetonInstruments.LightField.AddIns;
using LightFieldAddInSamples;

namespace LightFieldAddInSamples
{
    ///////////////////////////////////////////////////////////////////////////
    //  Acquire multiple frames and show the meta data for each frame    
    ///////////////////////////////////////////////////////////////////////////
    [AddIn("Metadata Sample (C#)",
    Version = "1.0.0",
    Publisher = "Teledyne Princeton Instruments", 
    Description="Demonstrates metadata tagging and how to access the timestamps on the data after acquisition on a frame by frame basis")]
    [QualificationData("IsSample", "True")]
    public class AddinMetaDataSample : AddInBase, ILightFieldAddIn
    {
        private LightFieldAddInSamples.Metadata_Sample.MetadataSample control_;
        
        ///////////////////////////////////////////////////////////////////////
        public UISupport UISupport { get { return UISupport.ExperimentSetting; } }
        ///////////////////////////////////////////////////////////////////////
        public void Activate(ILightFieldApplication app)
        {
            // Capture Interface
            LightFieldApplication = app;

            // Build your controls            
            control_ = new LightFieldAddInSamples.Metadata_Sample.MetadataSample(LightFieldApplication);
            ExperimentSettingElement = control_;
        }                        
        ///////////////////////////////////////////////////////////////////////
        public void Deactivate() { }
        ///////////////////////////////////////////////////////////////////////
        public override string UIExperimentSettingTitle { get { return "Metadata Sample"; } }
    }
}


